pulse-binding/src/context/introspect.rs

```diff
@@ -452,6 +462,27 @@ impl<'a> SinkInfo<'a> {
             }
         }
     }
+
+    /// Creates owned data from borrowed data
+    pub fn to_owned(&self) -> SinkInfo<'static> {
+        SinkInfo {
+            name: self.name.clone().map(|o| Cow::Owned(o.into_owned())),
+            description: self.description.clone().map(|o| Cow::Owned(o.into_owned())),
+            monitor_source_name: self
+                .monitor_source_name
+                .clone()
+                .map(|o| Cow::Owned(o.into_owned())),
+            driver: self.driver.clone().map(|o| Cow::Owned(o.into_owned())),
+            proplist: self.proplist.clone(),
+            ports: self.ports.iter().map(SinkPortInfo::to_owned).collect(),
+            active_port: self
+                .active_port
+                .as_ref()
+                .map(|spi| Box::new(spi.as_ref().to_owned())),
+            formats: self.formats.clone(),
```

Comment:
Same issue again. `format::Info` contains both a `Proplist` and also a raw inner proplist pointer. So again we'd have to use iterative string cloning and modify both `format::Info` and the `formats` attribute here to be able to hold either a borrowed or owned copy of the data.

Reply:
Similarly here, is seems safe but I don't know the internals of `pa_format_copy_info`
```rust
impl Clone for Info {
    /// Returns a new `Info` struct and representing the same format. If this is called on a ‘weak’
    /// instance, a non-weak object is returned.
    fn clone(&self) -> Self {
        let ptr = unsafe { capi::pa_format_info_copy(self.ptr as *const capi::pa_format_info) };
        assert_eq!(false, ptr.is_null());
        Self::from_raw(ptr as *mut InfoInternal)
    }
}
```